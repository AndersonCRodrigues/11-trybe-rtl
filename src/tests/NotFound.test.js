import { screen } from '@testing-library/react';
import { NotFound } from '../pages';
import renderWithRouter from '../renderWithRouter';

describe('', () => {
  it('testa se um caminho errado leva para "Not Found"', () => {
    renderWithRouter(<NotFound />);

    const heading = screen.getByRole('heading', { name: 'Page requested not found', level: 2 });
    const image = screen.getByRole('img', { name: 'Pikachu crying because the page requested was not found' });
    const imagemSrc = 'https://media.giphy.com/media/kNSeTs31XBZ3G/giphy.gif';

    expect(heading).toBeInTheDocument();
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute('src', imagemSrc);
  });

  it('mesma snapshot', () => {
    const { container } = renderWithRouter(<NotFound />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
