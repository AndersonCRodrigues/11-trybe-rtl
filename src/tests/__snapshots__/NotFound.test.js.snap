// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` mesma snapshot 1`] = `
<div
  class="not-found"
>
  <h2>
    Page requested not found
  </h2>
  <img
    alt="Pikachu crying because the page requested was not found"
    class="not-found-image"
    src="https://media.giphy.com/media/kNSeTs31XBZ3G/giphy.gif"
  />
</div>
`;
